Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10986,15 +11009,45 @@ def events_error_handler(res):
 
     zipped_data = gzip.compress(data.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
     client = BaseClient(base_url=xsiam_url)
+    data_chunks = split_xsiam_events_to_chunks(data, sys.getsizeof(zipped_data))
+    for data_chunk in data_chunks:
+        amount_of_events = len(data_chunk)
+        data_chunk = '\n'.join(data_chunk)
+        zipped_data = gzip.compress(data_chunk.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
+        send_to_xsiam_with_retries(client, events_error_handler, header_msg, headers, num_of_attempts, xsiam_url,
+                                   zipped_data)
+        demisto.updateModuleHealth({'eventsPulled': amount_of_events})
```

Comment:
This should actually be called just once after the for loop with the sum total of events created.